.github/workflows/test-runner-perf.yml

```diff
@@ -0,0 +1,47 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Run Test Runner Tests
+
+on:
+  push:
+    paths-ignore:
+      - 'README.md'
+      - 'local-dev/**'
+
+jobs:
+  test-runner-perf:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Get Vault token
+        id: vault-token-step
+        env:
+          VAULT_ADDR: https://clotho.broadinstitute.org:8200
+        run: |
+          VAULT_TOKEN=$(docker run --rm --cap-add IPC_LOCK \
+            -e "VAULT_ADDR=${VAULT_ADDR}" \
+            vault:1.1.0 \
+            vault write -field token \
+              auth/approle/login role_id=${{ secrets.VAULT_APPROLE_ROLE_ID }} \
+              secret_id=${{ secrets.VAULT_APPROLE_SECRET_ID }})
+          echo ::set-output name=vault-token::$VAULT_TOKEN
+          echo ::add-mask::$VAULT_TOKEN
+      - name: "Configure Test Runner"
+        run: |
+          cd ${GITHUB_WORKSPACE}/${workingDir}/workspace-manager-clienttests
+          chmod +x render-config.sh
+          ./render-config.sh ${{ steps.vault-token-step.outputs.vault-token }}
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            .gradle/caches
+            .gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle') }}
+          restore-keys: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}
+      - name: Run perf tests using test runner library
+        run: |
+          cd ${GITHUB_WORKSPACE}/${workingDir}/workspace-manager-clienttests
+          ./gradlew runTest --args="configs/BasicAuthenticated.json /tmp/TestRunnerResults"
```

Comment:
Two comments here:
- I think we need to add a step to build the client library from the current branch's code. Otherwise, it's testing new code with an old library. That will cause confusion when changes affected the client library.
- I think this GH action should be in a separate PR. In particular, we should add this after or concurrently with the uploadResults Gradle task. Otherwise, the tests will run in this GH action runner and then we lose the results.

Reply:
I have created a separate PR for this: [](https://github.com/DataBiosphere/terra-workspace-manager/pull/163).

Reply:
It looks like it's still included in this one though?

Reply:
I think this is all set. Nothing more needs to be done here.